Bug #338302: Use pkg-config instead of '{apr,aprutil,neon}-config --libs'
to improve link stuff.  At the time this patch was originally deployed,
it eliminated redundant dependencies on: libssl0.9.8, libldap2,
libexpat2, libxml2, zlib1g.

Index: git/build/ac-macros/apr.m4
===================================================================
--- git.orig/build/ac-macros/apr.m4	2013-11-03 20:02:08.000000000 +0100
+++ git/build/ac-macros/apr.m4	2013-11-03 20:02:43.000000000 +0100
@@ -94,9 +94,9 @@
       AC_MSG_ERROR([apr-config --link-ld --libs failed])
     fi
   else
-    SVN_APR_LIBS="`$apr_config --link-ld`"
+    SVN_APR_LIBS="`pkg-config --libs apr-1`"
     if test $? -ne 0; then
-      AC_MSG_ERROR([apr-config --link-ld failed])
+      AC_MSG_ERROR([pkg-config --libs apr-1 failed])
     fi
   fi
   SVN_APR_LIBS="`SVN_REMOVE_STANDARD_LIB_DIRS($SVN_APR_LIBS)`"
Index: git/build/ac-macros/aprutil.m4
===================================================================
--- git.orig/build/ac-macros/aprutil.m4	2013-11-03 20:02:08.000000000 +0100
+++ git/build/ac-macros/aprutil.m4	2013-11-03 20:02:43.000000000 +0100
@@ -98,9 +98,9 @@
       AC_MSG_ERROR([apu-config --link-ld --libs failed])
     fi
   else
-    SVN_APRUTIL_LIBS="`$apu_config --link-ld`"
+    SVN_APRUTIL_LIBS="`pkg-config --libs apr-util-1`"
     if test $? -ne 0; then
-      AC_MSG_ERROR([apu-config --link-ld failed])
+      AC_MSG_ERROR([pkg-config --libs apr-util-1 failed])
     fi
   fi
   SVN_APRUTIL_LIBS="`SVN_REMOVE_STANDARD_LIB_DIRS($SVN_APRUTIL_LIBS)`"
